
// The test succeeds when run in interpreter mode. However, it fails 
// when run in compilation mode. The reason behind that migth be
// the fact that there's only one output file that catches the error 
// in interpreter. Yet, the error is not catched for the compiler.
// A possible solution is to make more div tests that test both the 
// interpreter and the compiler.


fun int mul(int x, int y) =
    x / y  

fun int testMul(int i) =
    if i == 0 then
         0
    else 
        let n1 = read(int) in  
        let n2 = read(int) in
        let result = write(mul(n1,n2)) in
        testMul(i-1)

fun int main() =
    let result = testMul(5) in
    0
    
   
    

